### PR TITLE
UX updates to timezone config

### PR DESCRIPTION
**Pull request by aferd**
A few UX updates from Margot...
1. Default time should be selected as "default" setting
2. Default time should be 00:00 UTC midnight
3. Align dropdowns with custom time vs. radio button

### CODE REVIEW

**Comment by aferd** [comment]
@josejulio the "save" is not getting hit and I think this is the last piece of this that needs work - can you review when you get a chance?

**Comment by aferd** [comment]
Screen Shot 2023-06-16 at 3 52 24 PM
Screen Shot 2023-06-16 at 3 53 10 PM

File: src/components/Notifications/TimeConfig.tsx [hunk 1]
```diff
@@ -73,65 +77,65 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
-                                            <Radio
-                                                isChecked={ radioSelect && !showCustomSelect }
-                                                onChange={ handleRadioSelect }
-                                                value={ timePref }
-                                                id='settings-time-config'
-                                                label='Default time'
-                                                description='12:00 UTC'
-                                                name='radio-select'>
-                                            </Radio>
+                                {/* {timePref ? */}
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: Is this supposed to be removed?

File: src/components/Notifications/TimeConfig.tsx [hunk 2]
```diff
@@ -73,65 +77,65 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
-                                    : <Skeleton />}
+                                                </DropdownToggle> }
+                                                isOpen={ isOpen }
+                                                onSelect={ handleTimeSelect }
+                                                menuAppendTo={ () => document.body }
+                                                dropdownItems={ dropdownItems }>
+                                            </Dropdown>
+                                        </StackItem></>)}
+                                </Stack>
+                                {/* : <Skeleton />} */}
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: Is this supposed to be removed?

**Comment by josejulio** [comment]
Missed this from other review, but https://github.com/RedHatInsights/notifications-frontend/blob/master/src/services/Notifications/GetTimePreference.ts#LL14C12-L14C23 should be `useQuery` instead of `useMutate`

**Comment by josejulio** [comment]
https://github.com/RedHatInsights/notifications-frontend/blob/master/src/components/Notifications/TimeConfig.tsx#L52

Here, the function receives `time`, but is not what you expect. The argument comes from the `onClick` function which is usually an event. 

I think you would need to use `handleTimeSelect` to store in a state the selected time and use that value in the function.

File: src/components/Notifications/TimeConfig.tsx [hunk 3]
```diff
@@ -45,14 +51,15 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
 
-    const handleTimeSelect = React.useCallback(() => {
+    const handleTimePrefSelect = React.useCallback(() => {
+        setTimeSelect(timeSelect);
         setIsOpen(false);
-    }, []);
+    }, [ timeSelect ]);
 
-    const handleButtonSave = React.useCallback((time) => {
+    const handleButtonSave = React.useCallback((timeSelect) => {
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]:
```suggestion
    const handleButtonSave = React.useCallback(() => {
```

Remove `timeSelect`, it's not a parameter sent to this function. This function only receives an event.
You want to use `timeSelect` from the outer scope.

Comment (josejulio) [reply src/components/Notifications/TimeConfig.tsx]: You will need to add `timeSelect` to the dependencies of ths `useCallback` function:
https://github.com/RedHatInsights/notifications-frontend/blob/f75e16d225d4ce0b9759f818aab8678bf3f1ddf3/src/components/Notifications/TimeConfig.tsx#L74

File: src/components/Notifications/TimeConfig.tsx [hunk 4]
```diff
@@ -73,71 +90,67 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
+                                            label='Default time'
+                                            description='00:00 UTC'
+                                            name='radio-select'>
+                                        </Radio>
+                                    </StackItem>
+                                    <StackItem>
+                                        <Radio
+                                            isChecked={ radioSelect && showCustomSelect }
+                                            onChange={ handleCustomRadioSelect }
+                                            id='settings-time-config'
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: We need to use a different id here.
```suggestion
                                            id='settings-time-config-custom'
```

File: src/components/Notifications/TimeConfig.tsx [hunk 5]
```diff
@@ -1,36 +1,52 @@
...
+    // Set the time preference value once we load it from the server
+    useEffect(() => {
+        if (timePref) {
+            setTimeSelect(timePref);
+        }
+    }, [ timePref ]);
 
-    const handleRadioSelect = React.useCallback(() => {
+    const handleRadioSelect = React.useCallback((time) => {
+        setTimeSelect(time);
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: What is this `time` value? Is this the default time? where does it come from?

File: src/components/Notifications/TimeConfig.tsx [hunk 6]
```diff
@@ -45,26 +61,27 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
-    }, []);
+    }, [ ]);
+
+    const handleButtonSave = React.useCallback(() => {
+        if (timeSelect) {
+            const mutate = saveTimePreference.mutate;
+            mutate({
+                body: timeSelect
+            }).then((response) => {
+                if (response.status === 200) {
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: This is changing to 204 (No content) - I'll prepare a followup PR to sync after we merge https://github.com/RedHatInsights/notifications-backend/pull/2069

File: src/components/Notifications/TimeConfig.tsx (hunk 5 above)
Comment (aferd) [reply src/components/Notifications/TimeConfig.tsx]: this was looking to save the default time, but looking again it should be body as the prop. If someone selects the default time, i was not sending that default time.

Comment (aferd) [reply src/components/Notifications/TimeConfig.tsx]: well, that is also not right... the body is returning a boolean...

Comment (aferd) [reply src/components/Notifications/TimeConfig.tsx]: Not needed as the value is being sent when "save" is hit.

File: src/components/Notifications/TimeConfig.tsx [hunk 7]
```diff
@@ -45,26 +60,27 @@ export const TimeConfigComponent: React.FunctionComponent = () => {
...
```
Comment (josejulio) [inline src/components/Notifications/TimeConfig.tsx]: Lets change this to 204 - the other PR has been merged
```suggestion
                if (response.status === 204) {
```

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**